CMakeLists.txt

```
@@ -2,7 +2,7 @@
 
 cmake_minimum_required(VERSION 3.15.0)
 
-if (APPLE AND CMAKE_VERSION VERSION_GREATER_EQUAL 4.0.0)
+if (APPLE AND CMAKE_VERSION VERSION_GREATER_EQUAL 4.0.0 AND NOT CMAKE_OSX_SYSROOT)
```

Nit: technically we do not need to check `CMAKE_VERSION` anymore.

```suggestion
if (APPLE AND NOT CMAKE_OSX_SYSROOT)
```

---

It's just to be extra sure, not reason to change anything on CMake < 4.